src/services/cipher.service.ts

```diff
@@ -903,12 +905,18 @@ export class CipherService implements CipherServiceAbstraction {
 
     async restoreWithServer(id: string): Promise<any> {
         await this.apiService.putRestoreCipher(id);
-        await this.restore(id);
+        const response = await this.apiService.getCipher(id);
+        await this.restore({ id: id, revisionDate: response.revisionDate });
     }
 
     async restoreManyWithServer(ids: string[]): Promise<any> {
         await this.apiService.putRestoreManyCiphers(new CipherBulkRestoreRequest(ids));
-        await this.restore(ids);
+        const restores: { id: string, revisionDate: string; }[] = [];
+        for (const id of ids) {
```

Comment:
this feels hacky to me. Should we just change the server signature to return the updated ciphers?

Reply:
I like that much better... shouldn't hurt reverse compatibility in that way since the client can ignore returned JSON (older client versions) but newer client versions can take advantage of it. Any time we're hitting an API (or database) in a loop it makes my skin crawl a bit

Reply:
The server changes are in bitwarden/server#1072